src/app/components/Options/Options.jsx

```diff
@@ -1,232 +1,110 @@
-    const account = {
-      name: values.name,
-      config: {
-        macaroon: values.macaroon,
-        url: values.url,
-      },
-      connector: "lnd",
-    };
-    return saveAccount(account).then(() => {
-      load();
-      if (formRef) {
-        formRef.resetFields();
-      }
-    });
-  };
-
-  const saveLndHubAccount = (values, formRef) => {
-    const account = {
-      name: values.name,
-      config: {
-        login: values.login,
-        password: values.password,
-        url: values.url,
-      },
-      connector: "lndhub",
-    };
-
-    return saveAccount(account).then(() => {
-      load();
-      if (formRef) {
-        formRef.resetFields();
-      }
-    });
-  };
-
-  const saveLnBitsAccount = (values, formRef) => {
-    const account = {
-      name: values.name,
-      config: {
-        adminkey: values.adminkey,
-        readkey: values.readkey,
-        url: "https://lnbits.com",
-      },
-      connector: "lnbits",
-    };
-
-    return saveAccount(account).then(() => {
-      load();
-      if (formRef) {
-        formRef.resetFields();
-      }
-    });
-  };
-
-  const saveNativeAccount = (values, formRef) => {
-    const account = {
-      name: values.name,
-      config: {},
-      connector: "native",
-    };
-    return saveAccount(account).then(() => {
-      load();
-      if (formRef) {
-        formRef.resetFields();
-      }
-    });
-  };
-
-  const resetAccounts = () => {
-    return accountsStore.reset();
-  };
-
-  const resetAllowances = () => {
-    return allowancesStore.reset();
-  };
-
-  const saveAccount = (account) => {
-    account.config = encryptData(account.config, password, settingsStore.salt);
-    return accountsStore.setAccount(account, true).then(() => {
-      return load();
-    });
-  };
-
-  const formSubmitFailure = (errorInfo) => {
-    console.log(errorInfo);
-  };
-
-  // TODO: refactor
-  const handlePasswordModalOk = (password) => {
-    setPassword(password);
-    if (Object.entries(accounts).length > 0) {
-      msg
-        .request("unlock", { password })
-        .then((response) => {
-          setShowPasswordModal(false);
-        })
-        .catch((e) => {
-          alert("Invalid password");
-        });
-    } else {
-      setShowPasswordModal(false);
+class Options extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+    this.history = createHashHistory();
+    this.checkDataStoreState();
+  }
+
+  async checkDataStoreState() {
+    await passwordSvc.checkPassword();
+    this.setState({ isInitialized: await passwordSvc.isInitialized() });
+    this.setState({ isUnlocked: await passwordSvc.isUnlocked() });
+    this.updateRoute();
+  }
+
+  async handlePasswordConfigured() {
+    await this.checkDataStoreState();
+  }
+
+  async handleUnlock() {
+    await this.checkDataStoreState();
+  }
+
+  async updateRoute() {
+    if (this.state.isInitialized === false) {
+      return this.history.replace("/init");
```

Comment:
If nothing has been configured we might want to send the user to welcome.html which includes the onboarding flow.

Reply:
agree